plugins/inputs/nsq/nsq.go

```diff
@@ -78,21 +110,54 @@ func (n *NSQ) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
-var tr = &http.Transport{
-	ResponseHeaderTimeout: time.Duration(3 * time.Second),
+func (n *NSQ) buildTLSConfig() (*tls.Config, error) {
```

Comment:
You can use `internal.GetTLSConfig` to do this: https://github.com/influxdata/telegraf/blob/master/internal/internal.go#L118